micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmThreadMetrics.java

```diff
@@ -97,4 +92,25 @@ private static String getStateTagValue(Thread.State state) {
         return state.name().toLowerCase().replace("_", "-");
     }
 
+    private Long getThreadStateCount(Thread.State state) {
+        Map<Thread.State, Long> stateCountGroup = threadStateGroupLocal.get();
+        Long count = stateCountGroup.remove(state);
+        if (stateCountGroup.isEmpty()) {
+            threadStateGroupLocal.remove();
+        }
+        return count;
+    }
+
+    private Map<Thread.State, Long> getThreadStatesGroup() {
+        ThreadMXBean threadBean = ManagementFactory.getThreadMXBean();
+        long[] allThreadIds = threadBean.getAllThreadIds();
+        Map<Thread.State, Long> stateCountGroup = Arrays.stream(threadBean.getThreadInfo(allThreadIds))
+                .collect(Collectors.groupingBy(ThreadInfo::getThreadState, () -> new EnumMap<>(Thread.State.class),
+                        Collectors.counting()));
+        for (Thread.State state : Thread.State.values()) {
+            stateCountGroup.putIfAbsent(state, null);
```

Comment:
Putting "null" will make the Gauge reporting NaN instead of "0" which I believe conveys better. And based on exporting systems this might be breaking when Gauge returns NaN.

Reply:
This is my problem, I misremembered the default value if there is no relevant state thread, have modified the code and committed